sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -226,6 +226,7 @@ class SessionCatalog(
           s"${globalTempViewManager.database}.viewName.")
     }
     requireDbExists(dbName)
+    externalCatalog.setCurrentDatabase(dbName)
```

Comment:
So far, everything works well without the fix. It only impacts the test cases, because when we always qualify the table names with the database names.

IMO, we should backport it to Spark 2.1.1.